packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -45,14 +45,11 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         afterDraw: (chartInstance: Chart, _easing: Chart.Easing, _options?: any) => { this.afterChartDraw(chartInstance); }
     };
     private updateSelection = (event: MouseEvent) => {
-        const scale = this.props.viewRange.getEnd() - this.props.viewRange.getstart();
         if (this.mouseIsDown && this.props.unitController.selectionRange && !this.isRightClick) {
             const xStartPos = this.props.unitController.selectionRange.start;
-            let end = xStartPos + ((event.screenX - this.posPixelSelect) / this.lineChartRef.current.chartInstance.width) * scale;
-            end = Math.min(Math.max(end, 0), this.props.unitController.absoluteRange);
             this.props.unitController.selectionRange = {
                 start: xStartPos,
-                end: end
+                end: this.getTimeX(event.offsetX)
```

Comment:
If you drag selection outside of the bounds of the XY chart then you start receiving event.offsetX in another component's coordinates and the end time is wrong.